File: Top 50 Coding Interview Questions/46 Write a Program to Find the Factorial of a Number using Recursion/FindtheFactorialofaNumberusingRecursion.cpp
#include <iostream>
using namespace std;

int main() {
    int num = 6, fact = 1;

    // Check if the number is negative
    if (num < 0) {
        cout << "Factorial of a negative number doesn't exist.\n";
    } else {
        // Calculate factorial for non-negative numbers
        for (int i = 1; i <= num; i++) {
            fact *= i; // Multiply current number with factorial
        }

        // Display the factorial
        cout << "Factorial of " << num << " is: " << fact << endl;
    }

    // Time complexity: O(N), where N is the input number
    // Space complexity: O(1), as only a constant amount of space is used

    return 0;
}

File: Top 50 Coding Interview Questions/46 Write a Program to Find the Factorial of a Number using Recursion/FindtheFactorialofaNumberusingRecursion.py
# Function to calculate factorial using recursion
def factorial(n):
    # Base case: factorial of 0 is 1
    if n == 0:
        return 1
    # Recursive case: n * factorial of (n-1)
    return n * factorial(n - 1)

# Input number
num = 5

# Display the result
print("Factorial of", num, "is", factorial(num))
